tests/phpunit/Mutator/MutatorFactoryTest.php

```diff
@@ -0,0 +1,334 @@
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Tests\Mutator;
+
+use function array_diff;
+use function array_values;
+use function count;
+use function get_class;
+use Infection\Mutator\Arithmetic\Minus;
+use Infection\Mutator\Arithmetic\Plus;
+use Infection\Mutator\Boolean\FalseValue;
+use Infection\Mutator\Boolean\IdenticalEqual;
+use Infection\Mutator\Boolean\NotIdenticalNotEqual;
+use Infection\Mutator\Boolean\TrueValue;
+use Infection\Mutator\MutatorFactory;
+use Infection\Mutator\ProfileList;
+use Infection\Mutator\Util\Mutator;
+use Infection\Visitor\ReflectionVisitor;
+use InvalidArgumentException;
+use PhpParser\Node;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use ReflectionClass;
+use function Safe\sprintf;
+
+final class MutatorFactoryTest extends TestCase
+{
+    /**
+     * @var MutatorFactory
+     */
+    private $mutatorFactory;
+
+    protected function setUp(): void
+    {
+        $this->mutatorFactory = new MutatorFactory();
+    }
+
+    public function test_it_creates_no_mutator_if_no_profile_or_mutator_is_passed(): void
+    {
+        $mutators = $this->mutatorFactory->create([]);
+
+        $this->assertCount(0, $mutators);
+    }
+
+    public function test_it_can_creates_the_mutators_for_a_given_profile(): void
+    {
+        $mutators = $this->mutatorFactory->create(['@boolean' => true]);
+
+        $this->assertSameMutatorsByClass(ProfileList::BOOLEAN_PROFILE, $mutators);
+    }
+
+    public function test_it_can_creates_the_mutators_with_empty_settings_for_a_given_profile(): void
+    {
+        $mutators = $this->mutatorFactory->create(['@boolean' => []]);
+
+        $this->assertSameMutatorsByClass(ProfileList::BOOLEAN_PROFILE, $mutators);
+    }
+
+    public function test_it_can_creates_the_profile_mutators_with_the_given_settings(): void
+    {
+        // TODO: refactor this test to check the mutator configuration directly instead of relying
```

Comment:
Is there any ticket for it? just to not leave it forever.

Same for other TODOs

Reply:
It's the next PR already created waiting for a rebase :)